src/lib/EventTypes.ts

```diff
@@ -0,0 +1,274 @@
+import { EventEmitter } from 'events'
+import {
+	BACnetMessage,
+	BACnetMessageHeader,
+	IAMResult,
+	WhoIsResult,
+	BACNetEventInformation,
+	BACNetAlarm,
+	DecodeAcknowledgeSingleResult,
+	DecodeAcknowledgeMultipleResult,
+	ServiceMessage,
+	SimpleAckPayload,
+	CovNotifyPayload,
+	AtomicFilePayload,
+	SubscribeCovPayload,
+	DeviceCommunicationControlPayload,
+	ReinitializeDevicePayload,
+	EventNotificationPayload,
+	ReadRangePayload,
+	ObjectOperationPayload,
+	ListElementOperationPayload,
+	PrivateTransferPayload,
+	RegisterForeignDevicePayload,
+	WhoHasPayload,
+	TimeSyncPayload,
+	IHavePayload,
+} from './types'
+
+export type Constructor<T = object> = new (...args: any[]) => T
+
+export function applyMixin(
+	target: Constructor,
+	mixin: Constructor,
+	includeConstructor = false,
+): void {
+	// Figure out the inheritance chain of the mixin
+	const inheritanceChain: Constructor[] = [mixin]
+
+	while (true) {
+		const current = inheritanceChain[0]
+		const base = Object.getPrototypeOf(current)
+		if (base?.prototype) {
+			inheritanceChain.unshift(base)
+		} else {
+			break
+		}
+	}
+	for (const ctor of inheritanceChain) {
+		for (const prop of Object.getOwnPropertyNames(ctor.prototype)) {
+			// Do not override the constructor
+			if (includeConstructor || prop !== 'constructor') {
+				Object.defineProperty(
+					target.prototype,
+					prop,
+					Object.getOwnPropertyDescriptor(ctor.prototype, prop) ??
+						Object.create(null),
+				)
+			}
+		}
+	}
+}
+
+/**
+ * Event types for BACnet client
+ */
+export interface BACnetClientEvents {
+	error: (error: Error) => void
+	listening: () => void
+	unhandledEvent: (content: ServiceMessage | BACnetMessage) => void
+	readProperty: (content: {
+		header?: BACnetMessageHeader
+		payload: DecodeAcknowledgeSingleResult
+	}) => void
```

Comment:
Create a generic type for this function as the only thing that changes the payload type

Reply:
```ts
export type ServiceResponse<T> = (content: {
		header?: BACnetMessageHeader
		payload: T
	}) => void
```